numpy/lib/function_base.py

```diff
@@ -232,7 +232,7 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
                 tmp_a = tmp_a[keep]
                 if tmp_w is not None:
                     tmp_w = tmp_w[keep]
-            tmp_a = np.array(tmp_a, dtype=float, copy=False)
+            tmp_a = np.array(tmp_a, dtype=float, copy=True)
```

Comment:
I think the intention of that line is better conveyed using the `astype` method, which defaults to `copy=True`, probably not worth setting it explicitly. Also, I have a feeling it is better to use `np.double` or `np.float64` rather than `float`, but perhaps @njsmith has some thoughts on the proper thing to use from the recent discussion on deprecating `np.float` and friends.